Rent.Service/Rent.Service.API/Controllers/ApiControllerBase.cs

```diff
@@ -0,0 +1,12 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Rent.Service.API.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public abstract class ApiControllerBase : ControllerBase
+{
+    private ISender _mediator;
+    protected ISender Mediator => _mediator ??= HttpContext.RequestServices.GetRequiredService<ISender>();
```

Comment:
Why don't you use constructor for that? Why do you need protected field incapsulating private?